### PR TITLE
Fix blocking clickevents

### PR DESCRIPTION
**Pull request by richese**
Added new function `execute` to run external programs without waiting for their output (#254).

Also some additions to `run_through_shell`:
* Arguments are separated automatically if shell is not enabled. This way you can simply pass the string with the command without using `command.split()` every time you want to run something without using shell.
* Exceptions are now logged. Just in case they ever actually happen.

### CODE REVIEW

**Comment by teto** [comment]
Cool !

File: i3pystatus/core/modules.py [hunk 1]
```diff
@@ -106,7 +107,7 @@ def split_callback_and_args(cb):
             if cb is not "run":
                 getattr(self, cb)(*args)
         else:
-            run_through_shell(cb, *args)
+            execute(cb, *args)
```
Comment (enkore) [inline i3pystatus/core/modules.py]: We probably want detach=True here and no *args - opinions?

**Comment by richese** [comment]
@enkore 
You need the args depending on how you defined the clickevent.
`on_leftclick = "urxvt -e htop"`  ->  `cb = "urxvt -e htop" args = []`
`on_leftclick = ["urxvt", "-e", "htop"]`  ->  `cb = "urxvt" args = ["-e", "htop"]`
The first variant should be sufficient (since you can always create a script for anything more complex) so I'll remove `*args` and update the docs to use only first variant for external commands.

I also realized that 'detached' does not work with parameters because of how i3-msg works so I'll need to look into that too.

**Comment by richese** [comment]
Everything that needed fixing should be good now. :)

I have rewritten a huge chunk of the Callbacks documentation so tell me if you like it this way, see any typos or other grammatical mistakes on my part.

File: i3pystatus/core/command.py [hunk 2]
```diff
@@ -7,13 +7,26 @@
...
+
+    :param command: A string or a list of strings containing the name and
+     arguments of the program.
+    :param enable_shell: If set ot `True` users default shell will be invoked
+     and given ``command`` to execute. The ``command`` should obviously be a
+     string since shell does all the parsing.
     """
 
+    if not enable_shell and not isinstance(command, list):
+        command = command.split()
```
Comment (rscholer) [inline i3pystatus/core/command.py]: Please use `shlex.split(command)` this ensures correct splitting for commands like:
`binary --long-option="more than one word"`.
`command.split()` would result in:
`['binary', '--long-option="more', 'than', 'one' 'word"']`,
whereas `shlex.split(command)` correctly results in
`['binary', '--long-option=more than one word']`.

Comment (rscholer) [inline i3pystatus/core/command.py]: You need to use shlex.split, or a command like this won't be correctly split: `app --long="1 2" a b`.

File: i3pystatus/core/command.py [hunk 3]
```diff
@@ -7,13 +7,26 @@
...
+    output is saved in one variable.
+
+    :param command: A string or a list of strings containing the name and
+     arguments of the program.
+    :param enable_shell: If set ot `True` users default shell will be invoked
+     and given ``command`` to execute. The ``command`` should obviously be a
+     string since shell does all the parsing.
     """
 
+    if not enable_shell and not isinstance(command, list):
```
Comment (rscholer) [inline i3pystatus/core/command.py]: You should also test for tuples, or maybe test for strings. Catching the raised AttributeError is also an option.

**Pull request merged**